core/src/main/scala/org/apache/spark/util/DirectByteBufferOutputStream.scala

```diff
@@ -80,6 +96,7 @@ private[spark] class DirectByteBufferOutputStream(capacity: Int) extends OutputS
   override def close(): Unit = {
     // Eagerly free the direct byte buffer without waiting for GC to reduce memory pressure.
     StorageUtils.dispose(buffer)
```

Comment:
I checked and `StorageUtils.close(null)` is a no-op, so `close()` should be idempotent 👍